thetis/shallowwater_eq.py

```diff
@@ -265,10 +267,23 @@ def get_bnd_functions(self, eta_in, uv_in, bnd_id, bnd_conditions):
             h_ext = self.depth.get_total_depth(eta_ext)
             area = h_ext*bnd_len  # NOTE using internal elevation
             uv_ext = funcs['flux']/area*self.normal
-        else:
-            raise Exception('Unsupported bnd type: {:}'.format(funcs.keys()))
+        if eta_ext is None or uv_ext is None:
+            raise Exception('Unsupported open bnd type: {:}'.format(funcs.keys()))
         return eta_ext, uv_ext
 
+    def is_open_bnd(self, bnd_funcs, bnd_id):
```

Comment:
I find "open boundary" to be a bit misleading here. It suggests to me an open ocean boundary, but could you not impose such conditions on a coast, too?

Reply:
Good point. Renamed it to "bnd_state_defined".